### PR TITLE
fixing how external test clean up docker images

### LINKED ISSUE #3299: Fix how external tests clean up Docker images
External test jobs have been temporarily disabled from upstream pipeline. However if build job is triggered directly (not by pipeline) with default test list ( "sanity.openjdk", "sanity.system","extended.system", "sanity.perf", "sanity.external", https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-linux-x64-openj9/configure) https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-linux-x64-openj9/1103/ , external tests job will be triggered. https://ci.adoptopenjdk.net/view/Test_external/job/Test_openjdk11_j9_sanity.external_x86-64_linux/515/console

External test jobs story has been changed to run as parallel ( or later dynamic?), the old default configuration doesn't set as parallel, which means the job will run all enabled external sanity jobs. However the task  [`docker_prune`](https://github.com/adoptium/aqa-tests/blob/master/external/build.xml#L54) will recursively run and delete all earlier generated docker images and failed the job. https://ci.adoptopenjdk.net/view/Test_external/job/Test_openjdk11_j9_sanity.external_x86-64_linux/515/console

To avoid the confusion and prepare for re-enabling external tests, two steps can help

1. Update ci-pipeline-jenkins job config_regeneration to be consistent with nightlyDefault. 
2. Regenerate external test jobs to be latest and corresponding update to  parallel or dynamic.

@LongyuZhang noticed the failure in both adoptium and openj9 and questioned if any changes in external tests caused the failure

### PR DESCRIPTION
**Pull request by LizyBbethy**
after re fixing  docker_prune 
Fixes #3299

### CODE REVIEW

**Comment by LizyBbethy** [comment]
Grinder link https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3411/console
@llxia @sophia-guo @ShelleyLambert

File: external/build.xml [hunk 1]
```diff
@@ -38,7 +38,7 @@
 		<mkdir dir="${DEST_EXTERNAL}" />
 	</target>
 
-	<target name="build" depends="dist">
+	<target name="build" depends="docker_prune">
```
[the remark this answers is not in the document]
Comment (LizyBbethy) [reply external/build.xml:41]: @llxia should i let depends again have both dist and dockerprune?

**Comment by sophia-guo** [comment]
As @llxia mentioned `docker_prune` task need to be added as  the depends target of `build`, so at the beginning of each run( build process) task `docker_prune` will be called and only be called once. 

`<target name="build" depends="dist,docker_prune">` [in line 41](urlhttps://github.com/adoptium/aqa-tests/blob/master/external/build.xml#L41)

**Comment by sophia-guo** [comment]
The failure in  https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3415/console is not related with your changes, which is the issue with external _custom.

**Comment by LizyBbethy** [comment]
latest grinder sanity.external https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3454/console
jacoco grinder passed https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3455/console

**Comment by smlambert** [comment]
@sophia-guo - if you are happy with this PR, and we raise an issue for the new problem (related to needing a default value for external_custom), then I think we can bring this in and follow up separately with the external_custom update.

**Comment by sophia-guo** [comment]
Rerun https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3457/console

**Review by sophia-guo — APPROVED** [review]
LGTM. Thanks @LizyBbethy

**Review by smlambert — APPROVED** [review]
Good to go!

**Pull request merged**